### PR TITLE
Added option interfaces to io functions for intellisense.

### PR DESCRIPTION
**Pull request by cojmeister**
In response to the issue I opened earlier today (#37 ):
-  Added interfaces to io.ts functions such that they fit the options that appear in the docstring, and in the default options
-  Used this interfaces to cast the types of the options of each function - using the `Partial` utility type.

Should I create a base options interface and for each function extend? If so, where can I find a good description of all possible common options? And then, of each individual option?

Open to comments and suggestions! Also my first contribuition!

### CODE REVIEW

File: polars/io.ts [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
-import {Readable, Stream} from "stream";
-import {concat} from "./functions";
-
+import { DataFrame, _DataFrame } from "./dataframe";
+import { isPath } from "./utils";
+import { LazyDataFrame, _LazyDataFrame } from "./lazy/dataframe";
+import { Readable, Stream } from "stream";
+import { concat } from "./functions";
+
+export interface readCsvOptions {
```
Comment (universalmind303) [inline polars/io.ts:9]: can we make these interfaces `PascalCase`.

**Review by universalmind303 — CHANGES_REQUESTED** [review]
One small comment about the interface naming, otherwise looks good.

File: polars/io.ts (hunk 1 above)
Comment (cojmeister) [reply polars/io.ts]: Done!

File: polars/io.ts [hunk 2]
```diff
@@ -76,77 +105,113 @@ class LineBatcher extends Stream.Transform {
...
+  pathOrBody: string | Buffer,
+  options?: Partial<ReadCsvOptions>
+): DataFrame;
 export function readCSV(pathOrBody, options?) {
+  options = { ...readCsvDefaultOptions, ...options };
   const extensions = [".tsv", ".csv"];
 
   if (Buffer.isBuffer(pathOrBody)) {
-    return _DataFrame(pli.readCsv(pathOrBody, {...readCsvDefaultOptions, ...options}));
+    return _DataFrame(pli.readCsv(pathOrBody));
```
Comment (universalmind303) [inline polars/io.ts:180]: the rust bindings require a `ReadCsvOptions` as a second argument. It looks like this got removed here which is causing the errors.

Comment (cojmeister) [reply polars/io.ts]: Thanks - must have forgotten to re-add it after making options a variable.
Fixed it

File: sample.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import * as pl from "./polars"
+
+pl.col("A").str
```
Comment (universalmind303) [inline sample.ts:3]: can you delete this test file. Otherwise, it looks ready to merge.

Comment (cojmeister) [reply sample.ts]: Sorry 😬

**Pull request merged**